packages/react-server/src/ReactFlightServer.js

```diff
@@ -4246,6 +4243,27 @@ function enqueueFlush(request: Request): void {
   }
 }
 
+function enqueueOnAllReady(request: Request): void {
```

Comment:
I think this is unnecessary indirection.

every time we decrement our tasks via retryTask or erroredTask we can check whether we hit size zero and if we do, synchronously call onAllReady then. It's basically just the place where we set task status to COMPLETED in retryTask and the very end of erroredTask.

There are a couple decrement pathways that we hit during aborting but those actually don't need to do this because we already call onAllReady at the end of the abort function if there were any abortable tasks.